### PR TITLE
VSCode Plugin support

### PR DESCRIPTION
**Pull request by dgraeber**
### Description:

Added support for plugins on VSCode server in user-apps.  The jupyter-hub user image bootstrap was also modified to link in plugins to the workspace on the container at start.
A scripts dir was also added to fetch and set up STS credentials in a profile to support AWS Toolkit - this should be removed once support for containers is implemented by that team

### Issue Reference URL

https://github.com/awslabs/aws-eks-data-maker/issues/<NUMBER>

----------
### Do not change content below. Mark applicable check boxes only.

Thank you for submitting a contribution to AWS Orbit Workbench.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a ISSUE associated with this PR?
- [ ] Does your PR title start with ISSUE-XXXX where XXXX is the issue number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [ ] Has your PR been rebased against the latest commit within the target branch (typically 'main')?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under Apache License 2.0?

### CODE REVIEW

File: images/jupyter-user/bootstrap.sh [hunk 1]
```diff
@@ -47,3 +47,11 @@ do
 done
 
 rm -fR /home/jovyan/.aws
+
```
Comment (abaror) [inline images/jupyter-user/bootstrap.sh:50]: I don't think  "/home/jovyan/.local/share/code-server/CachedExtensionVSIXs" should ever exists on startup as this is a clean install.  Anyway , this check does not hurt I guess

File: samples/notebooks/M-Admin/extra_user_apps/apps/codeserver_proxy/install.sh [hunk 2]
```diff
@@ -20,4 +20,10 @@ curl -fsSL https://code-server.dev/install.sh | sh
 code-server --install-extension ms-python.python --force
 code-server --install-extension ms-toolsai.jupyter --force
 code-server --install-extension mtxr.sqltools --force
-pip install .
\ No newline at end of file
+code-server --install-extension extension-vsix/mtxr.sqltools-driver-pg-0.2.0.vsix --force
```
Comment (abaror) [inline samples/notebooks/M-Admin/extra_user_apps/apps/codeserver_proxy/install.sh:23]: here you can pull the binaries if need to , from the web.

File: samples/notebooks/M-Admin/extra_user_apps/apps/codeserver_proxy/scripts/getcreds.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env python3
```
Comment (abaror) [inline samples/notebooks/M-Admin/extra_user_apps/apps/codeserver_proxy/scripts/getcreds.py:1]: where is this file called from ?

**Review by abaror — CHANGES_REQUESTED** [review]
Why did we added binary files (the vsx plugins) into our sample notebook.  These should be fetched from the web as part of the docker build and not stored in our git repo

File: samples/notebooks/M-Admin/extra_user_apps/apps/codeserver_proxy/scripts/getcreds.py (hunk 3 above)
Comment (dgraeber) [reply samples/notebooks/M-Admin/extra_user_apps/apps/codeserver_proxy/scripts/getcreds.py:1]: It is a helper function to get credentials onto the instance to support AWS Toolkit.  It is not invoked automatically by any build process

File: samples/notebooks/M-Admin/extra_user_apps/apps/codeserver_proxy/install.sh (hunk 2 above)
Comment (dgraeber) [reply samples/notebooks/M-Admin/extra_user_apps/apps/codeserver_proxy/install.sh:23]: I was not able to pull the install for any drivers while on the console or during the build.  The vsix files are definitions, not binaries.  I see no other way to get the drivers installed than to store the vsix files of the binaries that we cannot install via pulling from the web in automation.

File: images/jupyter-user/bootstrap.sh (hunk 1 above)
Comment (dgraeber) [reply images/jupyter-user/bootstrap.sh:50]: Other modules use this bootstrap so the check is necessary

File: samples/notebooks/M-Admin/extra_user_apps/apps/codeserver_proxy/install.sh (hunk 2 above)
Comment (dgraeber) [reply samples/notebooks/M-Admin/extra_user_apps/apps/codeserver_proxy/install.sh:23]: I will try one more thing....from the VS Marketplace, the description is "This package is part of vscode-sqltools extension."-- perhaps mtxr.vscode-sqltools.sqltools-driver-pg is the full name of the extension??   If it fails, I suggest leaving the vsix files in the repo.

File: images/jupyter-user/bootstrap.sh (hunk 1 above)
Comment (dgraeber) [reply images/jupyter-user/bootstrap.sh:50]: Looks like other people have the same issue:  https://github.com/cdr/code-server/issues/1140 (last comment)

File: samples/notebooks/M-Admin/extra_user_apps/apps/codeserver_proxy/install.sh (hunk 2 above)
Comment (dgraeber) [reply samples/notebooks/M-Admin/extra_user_apps/apps/codeserver_proxy/install.sh:23]: Working on a resolution to embed the redshift driver into the image

Comment (dgraeber) [reply samples/notebooks/M-Admin/extra_user_apps/apps/codeserver_proxy/install.sh]: All binaries removed

File: samples/notebooks/M-Admin/extra_user_apps/apps/codeserver_proxy/scripts/getcreds.py (hunk 3 above)
Comment (dgraeber) [reply samples/notebooks/M-Admin/extra_user_apps/apps/codeserver_proxy/scripts/getcreds.py]: This is now invoked at bootstrap of the image

File: images/jupyter-user/bootstrap.sh (hunk 1 above)
Comment (dgraeber) [reply images/jupyter-user/bootstrap.sh:50]: Modified to look for both the existence of the source dir and the symlinks

Comment (dgraeber) [reply images/jupyter-user/bootstrap.sh:50]: Amir- you re correct.  I am moving this to the start script for codeserver...and reverting on next commit

**Review by abaror — APPROVED** [review]

**Pull request merged**